pkg/imagestream/imagestream_test.go

```diff
@@ -0,0 +1,350 @@
+package imagestream
+
+import (
+	"fmt"
+	"testing"
+
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	imagev1 "github.com/openshift/api/image/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	imageStreamName      = "network-tools"
+	imageStreamNamespace = "openshift"
+	testSchemes          = []clients.SchemeAttacher{
+		imagev1.AddToScheme,
+	}
+)
+
+func TestImageStreamPull(t *testing.T) {
+	generateImageStream := func(name, namespace string) *imagev1.ImageStream {
+		return &imagev1.ImageStream{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: namespace,
+			},
+		}
+	}
+
+	testCases := []struct {
+		name                string
+		namespace           string
+		addToRuntimeObjects bool
+		expectedError       error
+		client              bool
+	}{
+		{
+			name:                imageStreamName,
+			namespace:           imageStreamNamespace,
+			addToRuntimeObjects: true,
+			expectedError:       nil,
+			client:              true,
+		},
+		{
+			name:                "",
+			namespace:           imageStreamNamespace,
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("imageStream 'name' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                imageStreamName,
+			namespace:           "",
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("imageStream 'nsname' cannot be empty"),
+			client:              true,
+		},
+		{
+			name:                imageStreamName,
+			namespace:           imageStreamNamespace,
+			addToRuntimeObjects: false,
+			expectedError: fmt.Errorf("imageStream object %s does not exist in namespace %s",
+				imageStreamName, imageStreamNamespace),
+			client: true,
+		},
+		{
+			name:                imageStreamName,
+			namespace:           imageStreamNamespace,
+			addToRuntimeObjects: true,
+			expectedError:       fmt.Errorf("imageStream 'apiClient' cannot be empty"),
+			client:              false,
+		},
+	}
+
+	for _, testCase := range testCases {
+		// Pre-populate the runtime objects
+		var runtimeObjects []runtime.Object
+
+		var testSettings *clients.Settings
+
+		testImageStream := generateImageStream(testCase.name, testCase.namespace)
+
+		if testCase.addToRuntimeObjects {
+			runtimeObjects = append(runtimeObjects, testImageStream)
+		}
+
+		if testCase.client {
+			testSettings = clients.GetTestClients(clients.TestClientParams{
+				K8sMockObjects:  runtimeObjects,
+				SchemeAttachers: testSchemes,
+			})
+		}
+
+		builderResult, err := Pull(testSettings, testCase.name, testCase.namespace)
+		// assert.Equal(t, testCase.expectedError, err)
```

Comment:
you should be able to uncomment this and then simplify the if statement below a bit to not check for errors